frontend/packages/knative-plugin/src/components/add/event-sinks/KafkaSinkSection.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import { SelectVariant, TextInputTypes } from '@patternfly/react-core';
+import * as fuzzy from 'fuzzysearch';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import FormSection from '@console/dev-console/src/components/import/section/FormSection';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
+import { SecretModel } from '@console/internal/models';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import {
+  InputField,
+  ResourceDropdownField,
+  SelectInputField,
+  SelectInputOption,
+} from '@console/shared';
+import { EVENT_SINK_KAFKA_KIND } from '../../../const';
+import { getBootstrapServers } from '../../../utils/create-eventsources-utils';
+import { strimziResourcesWatcher } from '../../../utils/get-knative-resources';
+
+interface KafkaSinkSectionProps {
+  title: string;
+  namespace: string;
+  fullWidth?: boolean;
+}
+
+const KafkaSinkSection: React.FC<KafkaSinkSectionProps> = ({ title, namespace, fullWidth }) => {
+  const { t } = useTranslation();
+  const memoResources = React.useMemo(() => strimziResourcesWatcher(namespace), [namespace]);
+  const { kafkas, kafkaconnections } = useK8sWatchResources<{
+    [key: string]: K8sResourceKind[];
+  }>(memoResources);
+
+  const [bootstrapServers, bsPlaceholder] = React.useMemo(() => {
```

Comment:
@invincibleJai Since we need bootstrapServers and placeholder for both the kafkasource/sink forms we can create a hook that can be used in both places instead having same code in two places.